src/spinegeneric/cli/check_data_consistency.py

```diff
@@ -56,70 +103,39 @@ def main():
     fieldnames, rows = read_tsv(args.path_in / "participants.tsv")
 
     # Compare subject list from participants.tsv and from sub-* folders
-    tsv_subj = set(r["participant_id"] for r in rows)
-    dir_subj = set(p.name for p in args.path_in.glob("sub-*") if p.is_dir())
-    for subj in dir_subj - tsv_subj:
-        print(f"Warning missing subject from participants.tsv: {subj}")
-    for subj in tsv_subj - dir_subj:
-        print(f"Warning missing data for subject listed in participants.tsv: {subj}")
+    if "participant_id" in fieldnames:
+        tsv_subj = set(r["participant_id"] for r in rows)
+        dir_subj = set(p.name for p in args.path_in.glob("sub-*") if p.is_dir())
+        for subj in sorted(dir_subj - tsv_subj):
+            print(f"Warning: participants.tsv: missing row for data folder '{subj}'")
+        for subj in sorted(tsv_subj - dir_subj):
+            print(f"Warning: participants.tsv: missing data folder for subject '{subj}'")
 
     # Check the presence of JSON sidecars
     for img_path in args.path_in.glob("sub-*/**/*.nii.gz"):
         json_path = img_path.with_suffix("").with_suffix(".json")
         if not json_path.exists():
-            print(f"Missing jsonSidecar: {json_path}")
-
-    # Checking participants.tsv contents
-    schema = Schema(
-        [
-            Column(
-                "participant_id",
-                [LeadingWhitespaceValidation(), TrailingWhitespaceValidation()],
-            ),
-            Column("sex", [InListValidation(["M", "F"])]),
```

Comment:
I'm curious about the change from `["M", "F"]` to `["M", "F", "O"]`. Is Other part of the data spec? :D

Reply:
Well, it wasn't part of the original spec in this project, but it's [part of BIDS](https://bids-specification.readthedocs.io/en/stable/glossary.html#objects.columns.sex). I figured it's ok to have a custom validator be more restrictive than the spec (by not allowing lowercase or full words), but we might as well be inclusive because a definite "other" is not the same thing as a missing data "n/a".